proto/frequenz/api/dispatch/v1/dispatch.proto

```diff
@@ -247,6 +247,9 @@ message RecurrenceRule {
 
   // On which month(s) of the year does the event occur
   repeated uint32 bymonths = 8;
+
+  // The end time in UTC
+  google.protobuf.Timestamp end_time = 9;
```

Comment:
If we are fine with breaking changes maybe it would make sense to put this closer to `count` and maybe mention more explicitly what it means.

Also, wouldn't it make sense to have a `oneof` for `count` and `end_time`? I think normally iCal lets you use only one of them. If we allow both, we probably should also comment about how they play together (what happens if the `count` goes beyond the `end_time`?).

Reply:
Breaking changes are fine while we are still on v0.x.x, yeah. Those are some good questions, @thomas-nicolai-frequenz what do you think? Should `count` and `end_time` be mutually exclusive?

Reply:
> Should count and end_time be mutually exclusive?

No. Either of the two can end the dispatch. Whatever comes first.

Reply:
I checked the spec and they are not allowed both simultaneously:

> The UNTIL or COUNT rule parts are OPTIONAL,
> but they MUST NOT occur in the same 'recur'.

https://datatracker.ietf.org/doc/html/rfc5545#section-3.3.10

I would also name the field here `until` instead of `end_time` to match iCal if we are using it as a model (I would only diverge if necessary).

Reply:
OK, both fine for me. I though a little more flexibility would be nice to have but again we can stick with the standard for now until we have diverging requirements.